packages/firestore/test/unit/specs/write_spec.test.ts

```diff
@@ -648,6 +648,17 @@ describeSpec('Writes:', [], () => {
       .expectNumOutstandingWrites(0);
   });
 
+  specTest('New writes are sent after write failure', [], () => {
+    return spec()
+      .userSets('collection/a', { v: 1 })
+      .failWrite(
+        'collection/a',
+        new RpcError(Code.FAILED_PRECONDITION, 'failure')
+      )
+      .userSets('collection/b', { v: 1 })
+      .writeAcks('collection/b', 2000);
+  });
```

Comment:
Err... just to confirm, is this related to this PR?

Reply:
Yeah, this test case fails if the client always just uses the last existing batch from IndexedDB, since both writes end up getting the same batch ID.